src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/dto/GitLabPersonalAccessTokenRequestDTO.java

```diff
@@ -0,0 +1,66 @@
+package de.tum.in.www1.artemis.service.connectors.gitlab.dto;
+
+import java.util.Date;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class GitLabPersonalAccessTokenRequestDTO {
+
+    @JsonProperty
+    private String name;
+
+    @JsonProperty("user_id")
+    private Integer userId;
+
+    @JsonProperty("scopes")
+    private String[] scopes;
+
+    @JsonProperty("expires_at")
+    private Date expiresAt;
+
+    public GitLabPersonalAccessTokenRequestDTO() {
+        // default constructor for Jackson
+    }
+
+    public GitLabPersonalAccessTokenRequestDTO(String name, Integer userId, String[] scopes) {
+        this.name = name;
+        this.userId = userId;
+        this.scopes = scopes;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Integer getUserId() {
+        return userId;
+    }
+
+    public void setUserId(Integer userId) {
+        this.userId = userId;
+    }
+
+    public String[] getScopes() {
+        return scopes;
+    }
+
+    public void setScopes(String[] scopes) {
+        this.scopes = scopes;
+    }
+
+    public Date getExpiresAt() {
+        return expiresAt;
+    }
+
+    public void setExpiresAt(Date expiresAt) {
+        this.expiresAt = expiresAt;
+    }
```

Comment:
Why do you need expires_at ? Seems to be not used :) (Or do you intent this to be always null for the payload?)

Reply:
@dfuchss I added it because then the Gitlab API is "complete", in case it is ever used in a different scenario, but it is not currently needed.

Would you prefer to remove it?

Reply:
No, If it's part of the gitlab API everything is fine :)